nifi-nar-bundles/nifi-cdc/nifi-cdc-mssql-bundle/nifi-cdc-mssql-processors/src/main/java/org/apache/nifi/cdc/mssql/MSSQLCDCUtils.java

```diff
@@ -0,0 +1,264 @@
+            "FROM cdc.captured_columns cc\n" +
+            "LEFT OUTER JOIN cdc.index_columns ic ON \n" +
+            "(ic.object_id = cc.object_id AND ic.column_name = cc.column_name)\n" +
+            "where cc.object_id=?\n" +
+            "ORDER BY cc.column_ordinal";
+
+    public String getLIST_CHANGE_TRACKING_TABLES_SQL(){
+        return LIST_CHANGE_TRACKING_TABLES_SQL;
+    }
+
+    public String getLIST_TABLE_COLUMNS(){
+        return LIST_TABLE_COLUMNS;
+    }
+
+    public String getCURRENT_TIMESTAMP(){
+        return "CURRENT_TIMESTAMP";
+    }
+
+    public List<MSSQLTableInfo> getCDCTableList(Connection con) throws SQLException, CDCException {
+        ArrayList<MSSQLTableInfo> cdcTables = new ArrayList<>();
+
+        try(final Statement st = con.createStatement()){
+            final ResultSet resultSet = st.executeQuery(getLIST_CHANGE_TRACKING_TABLES_SQL());
+
+            while (resultSet.next()) {
+                int objectId = resultSet.getInt("object_id");
+                String databaseName = resultSet.getString("databaseName");
+                String schemaName = resultSet.getString("schemaName");
+                String tableName = resultSet.getString("tableName");
+                String sourceSchemaName = resultSet.getString("sourceSchemaName");
+                String sourceTableName = resultSet.getString("sourceTableName");
+
+                MSSQLTableInfo ti = new MSSQLTableInfo(databaseName, schemaName, tableName, sourceSchemaName, sourceTableName, Integer.toUnsignedLong(objectId), null);
+                cdcTables.add(ti);
+            }
+
+            for (MSSQLTableInfo ti:cdcTables) {
+                List<ColumnDefinition> tableColums = getCaptureColumns(con, ti.getTableId());
+
+                ti.setColumns(tableColums);
+            }
+        }
+
+        return cdcTables;
+    }
+
+    public List<ColumnDefinition> getCaptureColumns(Connection con, long objectId) throws SQLException, CDCException {
+        ArrayList<ColumnDefinition> tableColumns = new ArrayList<>();
+        try(final PreparedStatement st = con.prepareStatement(getLIST_TABLE_COLUMNS())){
+            st.setLong(1, objectId);
+
+            final ResultSet resultSet = st.executeQuery();
+            while (resultSet.next()) {
+                String columnName = resultSet.getString("column_name");
+                int columnId = resultSet.getInt("column_id");
+                String columnType = resultSet.getString("column_type");
+                int columnOrdinal = resultSet.getInt("column_ordinal");
+                int isColumnKey = resultSet.getInt("key");
+
+                int jdbcType = TranslateMSSQLTypeToJDBCTypes(columnType);
+
+                //get column list
+                MSSQLColumnDefinition col = new MSSQLColumnDefinition(jdbcType, columnName, columnOrdinal, isColumnKey==1);
+                tableColumns.add(col);
+            }
+        } catch (SQLException e) {
```

Comment:
You don't really need this catch block to rethrow the same exception, maybe cast it to CDCException or replace the catch block with an empty finally block?

Reply:
Agreed.